### ISSUE TITLE
media-posting/embedding

### ISSUE THREAD

**Issue by larsjaas** [RosenborgSupporterSoftware/RBKweb-ext#1]
Tenker for posting.php så kan man ha funksjonalitet for å gjenkjenne youtube og streamable o.l. linker og automatisk tilby embedding-kode med korrekt bredde osv.

**Comment by havremunken** [comment]
Jeg har også tenkt på dette, og synes dette er en god idé.

Men hva om vi i trådene istedenfor finner linker vi kan gjøre noe med, og rewriter posten til å auto-embedde?

Klart, din løsning vil da også hjelpe de som ikke har ext. når de leser, det vil ikke min.

**Comment by larsjaas** [comment]
Begge deler, og mulighet for revers overstyring for brukeren høres jo mest snasent ut.

**Comment by havremunken** [comment]
Det kan jo ta to former - en helper som merker støttede linker i poster og tilbyr å skrive dem om (eller gjør det automatisk, f.eks. før submit), og en som bare gjør om linker til visse sites til embed-objekter. Med tilhørende konfig-opsjoner.

Begge steder kan jo trekke fra samme base for å vite hva man kan embedde. Youtube er lett å begynne med, så legger man til støtte for vimeo eller soundcloud så støttes det automatisk begge steder f.eks.

**Comment by havremunken** [comment]
Yep, det kan jo lett være en opsjon ev. - Hva gjør vi med embeddable media?

- Ingenting (la standard link være i fred)
- Knapp for å embedde og bytte ut linken med objekt
- Auto-embedde

Og ev. andre ting som man ev. kommer på etterhvert.

**Comment by larsjaas** [comment]
Da har vi automatisk youtube-player-injisering i post-body'er via commit 8f9d35c, for å teste konseptet.

**Comment by larsjaas** [comment]
og tweets

**Comment by larsjaas** [comment]
og streamable

**Comment by larsjaas** [comment]
og jaggu er instagram på plass i 1. versjon også...

**Comment by havremunken** [comment]
Helt konge. :D

**Comment by larsjaas** [comment]
TODO:
- legg inn <video>-tag om non linker opp en .mp4 e.l.
- SoundCloud støtter oembed, så lag en kobling der.
- Twitter støtter oEmbed, så fjern iframe-teknikken til fordel for twitters oembed API.

**Comment by larsjaas** [comment]
TODO
- feature enable/disable for å reskalere eksisterende youtube-playere i post-bodyer som er satt opp for stort (bredt).
